tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -113,6 +123,32 @@ private function assertEquivalentCollections(Collection $expectedCollection, Col
         }
     }
 
+    private function checkServerlessRequirement(?string $serverless): void
+    {
+        switch ($serverless) {
+            case null:
+            case self::SERVERLESS_ALLOW:
+                return;
+
+            case self::SERVERLESS_FORBID:
+                if ($this->isServerless()) {
```

Comment:
I assume `checkServerlessRequirement` was intentionally duplicated in various spec test runners since it's more particular to the test runner, vs. `isServerless` which is more generic (like other methods we have to detect topologies or server versions).

Reply:
I just realized that CrudSpecFunctionalTest apperas to be unique in being ony of the only test runners (outside of unified and perhaps GridFS) not in the SpecTests directory. That explains why this doesn't share the same method in the base class in SpecTests.

Reply:
Yep - I opted against building something resilient or refactoring (we could theoretically use the unified test runner's `RunOnRequirement` class to handle this in all spec test runners), as those tests are migrated to the unified format in less time.